cpp/src/Slice/Parser.cpp

```diff
@@ -1275,9 +1279,7 @@ Slice::Container::createClassDef(const string& name, int id, const ClassDefPtr&
         return 0;
     }
 
-    bool isIllegal = !checkIdentifier(name);
-    isIllegal |= !checkForGlobalDef(name, "class");
-    if(isIllegal)
+    if(!checkIdentifier(name) | !checkForGlobalDef(name, "class"))
```

Comment:
We want to perform both checks, using `||` will short-circuit if the identifier syntax is invalid, and we won't perform the second check then. Using `|` will prevent short-circuiting and ensures both checks are performed as desired.

Reply:
Is it really possibly for both checks to fail?
If it is and we want to use `|` instead of `||` to always run both, then this definitely requires a comment.

Reply:
Yes, they're unrelated. One checks for illegeal (or reserved) syntax, the other checks if a slice construct has been defined in global scope (instead of module scope) I think it's better to issue errors for both simultaneously, instead of hiding one in the case both fail.

But I think you're correct about them needing a comment. I'll add one to clarify!